frontend/src/lib/getConvertedValue.ts

```diff
@@ -268,7 +270,19 @@ export function convertValue(
 	currentUnit: string,
 	targetUnit: string,
 ): number | null {
-	if (targetUnit === 'none') {
+	if (
+		targetUnit === 'none' ||
+		!currentUnit ||
+		!targetUnit ||
+		currentUnit === targetUnit
+	) {
+		if (
```

Comment:
Are we handling `percentage`? What about other units ?

Reply:
The only problem was for `percentage` unit, every other unit is working fine.

Reply:
I think it wasn't working for other units as well. Let me check and share video.